lib/sycamore/sycamore/query/execution/sycamore_executor.py

```diff
@@ -257,12 +258,24 @@ def get_code_string(self):
 """
         return result
 
+    def _write_metadata_to_trace_dir(self, plan: LogicalPlan, query_id: str):
+        assert self.trace_dir is not None, "Writing metadata requires trace_dir to be set"
+        path = os.path.join(self.trace_dir, f"{query_id}/metadata/")
+        os.makedirs(path, exist_ok=True)
+        with open(os.path.join(path, "query_plan.pickle"), "wb") as f:
+            pickle.dump(plan, f)
+
     def execute(self, plan: LogicalPlan, query_id: Optional[str] = None) -> Any:
         try:
             """Execute a logical plan using Sycamore."""
             if not query_id:
                 query_id = str(uuid.uuid4())
             bind_contextvars(query_id=query_id)
+
+            log.info("Writing query metadata")
```

Comment:
Say `metadata` one more time ...

Let's call this just `query_plan.json`. I don't think we need to keep using the term `metadata`.

Reply:
```
{
  "metadata": {
    "comment":
      {"response": "ack"}
    }
}
```